### DISCUSSION TITLE
Highlighting on search result page

### DISCUSSION THREAD

**Discussion by madjxatw** [sphinx-doc/sphinx#11951] [Q&A]
Did anybody ever encounter an issue that the matched words are not highlighted on the search result page? 

The docs are launched by executing `python3 -m http.server` from inside `_build/html` directory, and then are accessed in the web browser through `http://127.0.0.1:8000`. Every time a search is done, none of matched words are highlighted on the search result page, however, the target page does highlight all matched words.

Another issue is that only `h1` headings are listed on the search result page, but I saw in other online sphinx docs that all headings down to the matched word are listed. For example, if the matched word belongs to a section under a `h2` heading, then `h2` heading and the parent `h1` heading should both listed on the search result page, however, on my machine only the `h1` heading is shown unless the `h2` heading itself contains the matched word.

I've switched among several themes, but the issue persists. BTW, i am using Sphinx 7.1.2.

**Comment by wlach** [comment]
> The docs are launched by executing python3 -m http.server from inside _build/html directory, and then are accessed in the web browser through http://127.0.0.1:8000. Every time a search is done, none of matched words are highlighted on the search result page, however, the target page does highlight all matched words.

I've been working through various issues with search recently, some of which might cause this problem. It would be helpful if you could provide a reproducible test case with public documentation, as seen in https://github.com/sphinx-doc/sphinx/issues/11943#issue-2117282241

> Another issue is that only h1 headings are listed on the search result page, but I saw in other online sphinx docs that all headings down to the matched word are listed.

If your docs are using readthedocs (as the [main sphinx docs](https://www.sphinx-doc.org) are), a different search engine will be used which probably doesn't have the same bugs/problems as local search.

> For example, if the matched word belongs to a section under a h2 heading, then h2 heading and the parent h1 heading should both listed on the search result page, however, on my machine only the h1 heading is shown unless the h2 heading itself contains the matched word.

This sounds a lot like #11943. If you have another reproducible test case, please let me know (or just file a ticket).

**Comment by electric-coder** [comment]
Maybe this thread should be moved to the issue tracker since apparently it's a bug and not properly a *discussion*?

**Comment by picnixz** [comment]
Closing the discussion (see #11955 instead).
